content/hardware/04.pro/boards/portenta-x8/tutorials/datalogging-iot/content.md

```diff
@@ -0,0 +1,529 @@
+
+![Setting up data sources in Grafana via its GUI.](assets/x8-data-logging-img_18.png)
+
+Now, let's go to the InfluxDB desktop. Go to "Data" and select the test bucket we created before; in the bucket data explorer, in the query editor, select the `test` bucket, then in `_measurement` select counter, in `_field` select value, and `WINDOW PERIOD` select last as shown in the image below:
+
+![Setting up a query in the data explorer of the InfluxDB desktop.](assets/x8-data-logging-img_19.png)
+
+Now, click on "Script Editor" and copy the generated script by InfluxDB; we are going to use this script in Grafana to retrieve information from the `test` bucket, as shown in the image below:
+
+![Setting up a query in the data explorer of the InfluxDB desktop.](assets/x8-data-logging-img_20.png)
+
+Now, in the Grafana GUI, create a new dashboard and add a new panel; in the configuration page, select as "Data source" the bucket we configured before, `Test Bucket`:
+
+![Setting up a dashboard in Grafana via its GUI.](assets/x8-data-logging-img_21.png)
+
+In the "Query inspector," paste the script we generated before with InfluxDB. You should now see data from the tests made earlier with the MQTT client: 
+
+![Setting up data visualization in a Grafana dashboard via its GUI.](assets/x8-data-logging-img_22.png)
+
+Let's change how data is visualized. Select "Visualizations" and then search for "Gauge" and select it. Now we should visualize data as a gauge in Grafana:
+
+![Visualizing data with a gauge in a Grafana dashboard.](assets/x8-data-logging-img_23.png)
+
+We can change the panel options, such as their title and description. Click on apply; we can now use the MQTT client described before to test the MQTT broker integration with Node-RED, InfluxDB, and Grafana. Remember first to deploy the flow in Node-RED, subscribe to the test topic, and publish any value in the MQTT client. Also, remember to change the dashboard time range and its refresh rate. 
+
+![Configured dashboard in Grafana.](assets/x8-data-logging-img_24.png)
+
+## Sending Data Using the MKR WiFi 1010 Board
+
+Now, it is time to test our entire data logging application. We will use an [MKR WiFi 1010](https://store.arduino.cc/products/arduino-mkr-wifi-1010); this board will periodically send the value of a counter to the Grafana dashboard via the local MQTT broker deployed in the X8.
+
+First, let's ensure we have the required drivers for the MKR WiFi 1010 installed. This can be done by navigating to **Tools > Board > Board Manager...**, here we need to look for the **Arduino SAMD boards (32-bits ARM Cortex M0+)** and install the latest available version. We also need to install the libraries we will use to send data from the MKR WiFi 1010 board to the data logging application via MQTT. Go to **Tools > Manage libraries...**, search for **ArduinoMqttClient** and **WiFiNINA** and install the latest available version of both libraries.

```

Comment:
We can link the how to install a core instead of explaining it here

Reply:
@marqdevx is this the link https://docs.arduino.cc/learn/starting-guide/cores?

Reply:
Yes, I would use it with the anchor 
https://docs.arduino.cc/learn/starting-guide/cores#how-to-install-an-arduino-core

Reply:
Ok, done!